exercises/sum-of-multiples/example.scm

```diff
@@ -0,0 +1,11 @@
+(use-modules (srfi srfi-1))
+
+(define (sum-of-multiples ints limit)
+  (let ([ints-0 (delete 0 ints)])
+    (if (null? ints-0)
+        0
+        (apply + (filter (lambda (n)
+                           (any (lambda (i)
+                                  (zero? (remainder n i)))
+                                ints-0))
+                         (cdr (iota limit)))))))
```

Comment:
A small nitpick: You actually don't need the `null?` check since `any` returns false given an empty list

Reply:
Yes, you're right.  It's just that if the list of integers is `()`, and the limit is huge, then I'll save the effort of not running `iota`.

Reply:
Or how about this?

```scheme
(use-modules (srfi srfi-1))

(define (sum-of-multiples ints limit)
  (apply +
         (apply lset-union = (map (lambda (i)
                                    (iota (floor (/ (1- limit) i)) i i))
                                  (delete 0 ints)))))
```

Reply:
Yeah, you're right that it's faster, but it's still `O(m*N)` in the size of the range `N` and ints `m`. If speed is the ultimate goal it probably makes more sense to use an array and sieve or write something using exclusion-inclusion.

`lset-union` is fun but might actually get worse performance, since I think it's `O(m*(N/d)^2)` where `d` indicates the size of the ints. You could use the fact that the lists of multipes are sorted to recover `O(m*N/d)` (i think).

Either way is not a huge deal and we can stick with what's currently here if you'd prefer?

Reply:
OK.  Perhaps, we should stick with the simpler implementation.

Reply:
Simple sounds good!